src/main/java/com/kamiroo/todomanager/service/ToDoService.java

```diff
@@ -104,12 +110,34 @@ public void updateTodoOnTitleAndDescription(Long todoId, String title, String de
         toDoRepository.updateTodoOnTitleAndDescription(todoId, title, description);
     }
 
+    public int deleteTodoWhereStatusClosed() {
```

Comment:
Można to zrobić tak jak Ty to tutaj zrobiłeś i nie jest to złe, ale trochę mało wydajne, ponieważ najpierw czytasz wszystkie ToDos, potem filtrujesz te które mają status closed a na końcu usuwasz je pojedynczo.
Poprawkę, którą bym z pewnością zaproponował, byłoby użycie metody z Repository, która potrafiłaby przyjąć np. listę (lub w innej formie) obiektów do usunięcia. Poszukaj sobie "deleteAllById" (i czym się różni od deleteAllByIdInBatch). Tą poprawkę warto by było zrobić.
Natomiast już czysto teoretycznie najszybszym rozwiązaniem byłoby napisanie własnej metody która robiłaby w Javie to samo co w SQL robi "delete from to_do t where t.id in (...)"